tips/034_gleaning.md

```diff
@@ -0,0 +1,252 @@
+class FocusedGleaning(GleaningProcessor):
+    @lilypad.trace()
+    def focused_improvement(self, product_name: str, focus_areas: list[str]) -> str:
+        """Gleaning with specific focus areas for targeted improvement"""
+        
+        current_description = generate_initial_description(product_name)
+        
+        for focus_area in focus_areas:
+            print(f"\n🎯 Focusing on: {focus_area}")
+            
+            # Validate with specific focus
+            validation = detailed_validation(product_name, current_description)
+            
+            # Filter feedback for current focus area
+            relevant_errors = [
+                error for error in validation.errors 
+                if focus_area.lower() in error.lower()
+            ]
+            
+            relevant_suggestions = [
+                suggestion for suggestion in validation.suggestions
+                if focus_area.lower() in suggestion.lower()
+            ]
+            
+            if relevant_errors or relevant_suggestions:
+                current_description = improve_description(
+                    current_description,
+                    relevant_errors or ["General improvement needed"],
+                    relevant_suggestions or ["Enhance this area"],
+                    focus_area
+                )
+                print(f"Improved for {focus_area}: {current_description[:100]}...")
+        
+        return current_description
+
+# Example usage showing progressive improvement
+def demo_gleaning():
+    product = "Wireless Bluetooth Headphones"
+    final_description, history = gleaning_process(product, max_iterations=3)
+    
+    print(f"\n{'='*60}")
+    print("FINAL RESULT")
+    print('='*60)
+    print(f"Product: {product}")
+    print(f"Final Description: {final_description}")
+    print(f"\nImprovement History:")
+    for step in history:
+        print(f"  • {step}")
+
+# Focused gleaning example
+def demo_focused_gleaning():
+    focused_gleaner = FocusedGleaning()
+    
+    focus_areas = ["technical specifications", "user benefits", "competitive advantages"]
+    result = focused_gleaner.focused_improvement("Gaming Laptop", focus_areas)
+    
+    print(f"Focused result: {result}")
+
+if __name__ == "__main__":
+    demo_gleaning()
+    print("\n" + "="*60)
+    demo_focused_gleaning()
+```
+
+**Why this approach works better:**
+
+- **Learning from Failures:** Each validation failure provides specific guidance for targeted improvements
+- **Progressive Quality:** Builds on partial progress instead of starting over, leading to better final results
+- **Efficient Token Usage:** Improves existing content rather than generating completely new responses each time
+
+### The Takeaway
+
+Gleaning transforms validation failures into iterative improvement opportunities, using error feedback to refine responses progressively. This pattern achieves higher quality outputs with fewer token costs than regeneration approaches.
```

Comment:
we should also call out that its more costly and should only be used for more complex things